### ISSUE TITLE
[Feature Request] python API to run a single kernel

### ISSUE THREAD

**Issue by SimonRelu** [microsoft/onnxruntime#16847]
### Describe the feature request

Often, we want to do preprocessing/postprocessing on our data. Ideally, this would also happen on the target device (f.e. cuda). Currently we use cupy for this but this has the downside that it uses JIT compilation which means that it's significantly slower in a lot of cases. Mainly in cases where the tensor size is not constant.

Other libraries like pytorch provide an interface to run certain operations like resizing/scaling/softmax/...:
```
x = torch.tensor([1, 2, 3, 4])
x = torch.unsqueeze(x, 0)
x = torch.nn.functional.softmax(x)
```

I was wondering if it would be possible to do something similar in onnxruntime?

### use case

I did find the prepost processing in onnxruntime-extensions: https://github.com/microsoft/onnxruntime-extensions/tree/main/onnxruntime_extensions/tools/pre_post_processing but this feels a bit limiting. For one, I'd want to be able to do the resize once and then used this resized tensor as input for multiple models

another more complex use case : I want to be able to crop a scan around the nonzero values. I believe this should be possible to do using existing onnx layers like `nonzero`. The only problem is that I also need to know where this region is. If I were to add this as a post processing step from onnxruntime-extensions I'd also have to add an aditional output to the model

**Comment by Craigacp** [comment]
[ONNXScript](https://github.com/microsoft/onnxscript) does this, though possibly not in the most efficient way.

**Comment by xadupre** [comment]
You need to build a model with only one operator doing what you need to do. One way to do it is:

* write the code you want to execute with torch
* export it as a model
* use onnxruntime to execute it

These steps are very similar to what a jit does. If you need a couple of functionalities, you may use onnx directly to create your own graphs: https://onnx.ai/onnx/intro/python.html. If you need something more automated and you don't want to implement everything, onnxscript or spox should make your life easier.

**Comment by SimonRelu** [comment]
Thank you for all the feedback. I'll close this issue now
